### PR TITLE
Add initial functionality

### PR DESCRIPTION
**Pull request by dav3r**
#  #

## 🗣 Description ##

This PR sets up the initial functionality for this project, enabling the creation of new AWS accounts via Control Tower.

## 💭 Motivation and context ##

As we seek to automate all the things, this removes one more manual step in our process.

## 🧪 Testing ##

I successfully created two Staging accounts using this Terraform code.

## ✅ Checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -4,30 +4,54 @@
...
+  description = "The AWS region to deploy into (e.g. us-east-1)."
+  default     = "us-east-1"
 }
 
-variable "aws_availability_zone" {
+variable "provisioned_product_name" {
   type        = string
-  description = "The AWS availability zone to deploy into (e.g. a, b, c, etc.)."
-  default     = "a"
+  description = "Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name. (e.g. Example_Account."
```
Comment (jsf9k) [inline variables.tf]:
```suggestion
  description = "Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name. (e.g. Example_Account)."
```

File: variables.tf [hunk 2]
```diff
@@ -4,30 +4,54 @@
...
 }
 
-variable "aws_region" {
-  type        = string
-  description = "The AWS region to deploy into (e.g. us-east-1)."
-  default     = "us-east-1"
+variable "tags" {
+  type        = map(string)
+  description = "Tags to apply to all AWS resources created"
```
Comment (jsf9k) [inline variables.tf]:
```suggestion
  description = "Tags to apply to all AWS resources created."
```

File: README.md [hunk 3]
```diff
@@ -51,41 +72,32 @@ No modules.
...
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| ami\_owner\_account\_id | The ID of the AWS account that owns the Example AMI, or "self" if the AMI is owned by the same account as the provisioner. | `string` | `"self"` | no |
-| aws\_availability\_zone | The AWS availability zone to deploy into (e.g. a, b, c, etc.). | `string` | `"a"` | no |
+| account\_email | The root email address to associate with the AWS account (e.g. admin@example.com). | `string` | n/a | yes |
+| account\_name | The name of the AWS account to provision (e.g. Example Account). | `string` | n/a | yes |
+| account\_org\_unit | The name of the AWS Organizational Unit under which the account resides (e.g. Sandbox). | `string` | n/a | yes |
 | aws\_region | The AWS region to deploy into (e.g. us-east-1). | `string` | `"us-east-1"` | no |
-| subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0). | `string` | n/a | yes |
+| provisioned\_product\_name | Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name. (e.g. Example\_Account. | `string` | n/a | yes |
```
Comment (jsf9k) [inline README.md]:
```suggestion
| provisioned\_product\_name | Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name. (e.g. Example\_Account). | `string` | n/a | yes |
```

File: README.md [hunk 4]
```diff
@@ -51,41 +72,32 @@ No modules.
...
+| sso\_email | The email address of the SSO user (e.g. john.doe@example.com).  This email address must already exist in AWS SSO. | `string` | n/a | yes |
+| sso\_first\_name | The first name of the SSO user (e.g. John). | `string` | n/a | yes |
+| sso\_last\_name | The last name of the SSO user (e.g. Doe). | `string` | n/a | yes |
+| tags | Tags to apply to all AWS resources created | `map(string)` | `{}` | no |
```
Comment (jsf9k) [inline README.md]:
```suggestion
| tags | Tags to apply to all AWS resources created. | `map(string)` | `{}` | no |
```

**Review by jsf9k — CHANGES_REQUESTED** [review]

File: README.md [hunk 5]
```diff
@@ -51,41 +72,32 @@ No modules.
...
 | Name | Description | Type | Default | Required |
 |------|-------------|------|---------|:--------:|
-| ami\_owner\_account\_id | The ID of the AWS account that owns the Example AMI, or "self" if the AMI is owned by the same account as the provisioner. | `string` | `"self"` | no |
-| aws\_availability\_zone | The AWS availability zone to deploy into (e.g. a, b, c, etc.). | `string` | `"a"` | no |
+| account\_email | The root email address to associate with the AWS account (e.g. admin@example.com). | `string` | n/a | yes |
+| account\_name | The name of the AWS account to provision (e.g. Example Account). | `string` | n/a | yes |
+| account\_org\_unit | The name of the AWS Organizational Unit under which the account resides (e.g. Sandbox). | `string` | n/a | yes |
 | aws\_region | The AWS region to deploy into (e.g. us-east-1). | `string` | `"us-east-1"` | no |
-| subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0). | `string` | n/a | yes |
+| provisioned\_product\_name | Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name. (e.g. Example\_Account). | `string` | n/a | yes |
```
Comment (jsf9k) [inline README.md]:
```suggestion
| provisioned\_product\_name | Name of the service catalog product that is provisioned. If not provided, defaults to a slugified version of the account name (e.g. Example\_Account). | `string` | n/a | yes |
```

**Review by jsf9k — APPROVED** [review]
One more.

File: README.md (hunk 5 above)
Comment (dav3r) [reply README.md]: The period in this case is after the closing paren, similar to the others, so I think it's ok here too?

Comment (jsf9k) [reply README.md]: Right, but there's also an extra period _before_ the parentheses.  That's my quibble.

Comment (dav3r) [reply README.md]: Gah, you are right- I mis-read the diff.  Thanks!

**Review by jsf9k — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -2,46 +2,67 @@
...
+- Administrative access to an AWS account that can run Control Tower.
+- [Terraform](https://www.terraform.io/) installed on your system.
+- An accessible AWS S3 bucket to store Terraform state
+  (specified in [`backend.tf`](backend.tf)).
+- An accessible AWS DynamoDB database to store the Terraform state lock
+  (specified in [`backend.tf`](backend.tf)).
+- A Terraform [variables](variables.tf) file customized for your new
+  AWS account, for example:
+
+  ```console
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
  ```hcl
```

File: README.md [hunk 7]
```diff
@@ -2,46 +2,67 @@
...
-## Examples ##
-
-- [Basic usage](https://github.com/cisagov/provision-aws-account/tree/develop/examples/basic_usage)
+1. Create an AWS profile called `provision-aws-account` (if you haven't
+   already done so) in your AWS configuration using the
+   "AWSAdministratorAccess" credentials (access key ID, secret access key,
+   and session token) as obtained from the AWS account where you want
+   Control Tower to run:
+
+   ```console
```
Comment (mcdonnnj) [inline README.md]:
```suggestion
   ```ini
```

**Review by mcdonnnj — APPROVED** [review]
Excellent work! I only have two syntax highlighting suggestions for fenced code blocks.

**Comment by dav3r** [comment]
> Excellent work! I only have two syntax highlighting suggestions for fenced code blocks.

Thanks, I always forget about those! 👍

**Pull request merged**